src/node/node_to_node.h

```diff
@@ -138,6 +139,7 @@ namespace ccf
 
     void initialize(NodeId self_id, const tls::Pem& network_pkey) override
     {
+      std::unique_lock<SpinLock> guard(lock);
```

Comment:
This guards assignment of `self`, but we don't have the same guard on later reads of it. So if it was possible to call `initialize` multiple times, then the value could change during execution of (eg) `create_channel`. I think we should convince ourselves that `initialize` is only called once in a node's lifetime, while the enclave is single-threaded, before any other calls to this object, and then we can remove this lock.